crates/binstalk/src/ops/resolve.rs

```diff
@@ -91,44 +91,60 @@ async fn resolve_inner(
+        _ => None,
+    };
 
-    let data = Arc::new(Data::new(
-        package_info.name.clone(),
-        package_info.version_str.clone(),
-        package_info.repo.clone(),
-    ));
+    let mut handles: Vec<(Arc<dyn Fetcher>, _)> = Vec::with_capacity(
+        desired_targets.len() * resolvers.len() * if binary_name.is_some() { 2 } else { 1 },
+    );
 
-    handles.extend(
-        resolvers
-            .iter()
-            .cartesian_product(desired_targets.into_iter().map(|(triple, target)| {
-                debug!("Building metadata for target: {target}");
+    let mut handles_fn = |data: Arc<Data>| {
+        handles.extend(
+            resolvers
+                .iter()
+                .cartesian_product(desired_targets.clone().into_iter().map(|(triple, target)| {
+                    debug!("Building metadata for target: {target}");
 
-                let target_meta = package_info.meta.merge_overrides(
-                    iter::once(&opts.cli_overrides).chain(package_info.overrides.get(target)),
-                );
+                    let target_meta = package_info.meta.merge_overrides(
+                        iter::once(&opts.cli_overrides).chain(package_info.overrides.get(target)),
+                    );
 
-                debug!("Found metadata: {target_meta:?}");
+                    debug!("Found metadata: {target_meta:?}");
 
-                Arc::new(TargetData {
-                    target: target.clone(),
-                    meta: target_meta,
-                    target_related_info: triple,
-                })
-            }))
-            .map(|(f, target_data)| {
-                let fetcher = f(
-                    opts.client.clone(),
-                    opts.gh_api_client.clone(),
-                    data.clone(),
-                    target_data,
-                    opts.signature_policy,
-                );
-                (fetcher.clone(), AutoAbortJoinHandle::new(fetcher.find()))
-            }),
-    );
+                    Arc::new(TargetData {
+                        target: target.clone(),
+                        meta: target_meta,
+                        target_related_info: triple,
+                    })
+                }))
+                .map(|(f, target_data)| {
+                    let fetcher = f(
+                        opts.client.clone(),
+                        opts.gh_api_client.clone(),
+                        data.clone(),
+                        target_data,
+                        opts.signature_policy,
+                    );
+                    (fetcher.clone(), AutoAbortJoinHandle::new(fetcher.find()))
+                }),
+        )
+    };
+
+    handles_fn(Arc::new(Data::new(
+        package_info.name.clone(),
+        package_info.version_str.clone(),
+        package_info.repo.clone(),
+    )));
+
+    if let Some(binary_name) = binary_name {
+        handles_fn(Arc::new(Data::new(
```

Comment:
You don't need to repeat this for every resolver.

quick install resolver expects crate-name, and passing binary-name would just add extra query/networking overhead.

Reply:
You need to add an additional if `let Some(resolver) = resolvers.iter().find(|resolver| resolver.fetcher_name() == "GhCrateMeta")`

You can combine it with the existing `binary_name` check into `if let (Some(..), Some(..)) = (binary_name, ...) { ... }`

Reply:
OK, adding a filter by `fetcher_name` - into the closure, as `resolvers` is a vec of functions.